go/vt/vtgate/engine/limit.go

```diff
@@ -35,9 +35,10 @@ var _ Primitive = (*Limit)(nil)
 
 // Limit is a primitive that performs the LIMIT operation.
 type Limit struct {
-	Count  evalengine.Expr
-	Offset evalengine.Expr
-	Input  Primitive
+	Count                evalengine.Expr
+	Offset               evalengine.Expr
+	RequireCompleteInput bool
```

Comment:
I understand what these field does by looking at the comments, but it is not apparent directly. Could we add a comment explaining what it is and why its required.